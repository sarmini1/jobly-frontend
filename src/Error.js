import { Alert } from "react-bootstrap";

/** Error
 *
 * Props:
 * - error: string
 *
 * State:
 * None
 *
 * Most components --> Error
 */
function Error({ error }) {
  return (
    <div className="Error">
      <Alert variant='danger'>
        <p>{error}</p>
      </Alert>
    </div>

  );
}

export default Error;
