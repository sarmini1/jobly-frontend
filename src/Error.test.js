import React from 'react';
import { render } from '@testing-library/react';
import Error from "./Error";

it('mounts without crashing', function () {
  render(
    <Error error={"test error text"} />
  );
});

it("matches snapshot", function () {
  const { container } = render(
    <Error error={"test error text"} />
  );
  expect(container).toMatchSnapshot();
});

it("renders expected error info", function () {
  const { container } = render(
    <Error error={"test error text"} />
  );
  expect(container).toContainHTML('<p>test error text</p>');
});