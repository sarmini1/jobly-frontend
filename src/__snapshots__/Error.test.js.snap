// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Error"
  >
    <div
      class="fade alert alert-danger show"
      role="alert"
    >
      <p>
        test error text
      </p>
    </div>
  </div>
</div>
`;
